dspy/primitives/prediction.py

```diff
@@ -90,6 +90,15 @@ def __ge__(self, other):
         elif isinstance(other, Prediction):
             return self.__float__() >= float(other)
         raise TypeError(f"Unsupported type for comparison: {type(other)}")
+    
+    def __eq__(self, other):
+        if isinstance(other, (float, int)):
+            return self.__float__() == other
+        elif isinstance(other, Prediction):
+            return self.__float__() == float(other)
```

Comment:
nit: shall we do `float(self) == float(other)` for consistency?

Reply:
I guess this should be consistent with how __ge__ or __le__ are implemented?